### PR TITLE
Remove component will receive props from application links

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Part of the work to resolve https://github.com/cerner/terra-core/issues/1884

### CODE REVIEW

File: packages/terra-application-links/src/tabs/ApplicationTabs.jsx [hunk 1]
```diff
@@ -80,13 +80,11 @@ class ApplicationTabs extends React.Component {
     this.handleResize(this.contentWidth);
   }
 
-  componentWillReceiveProps(newProps) {
```
Comment (dkasper-was-taken) [inline packages/terra-application-links/src/tabs/ApplicationTabs.jsx]: We'll want to look at either using shouldComponentUpdate or moving this check to the render method.

Comment (bjankord) [reply packages/terra-application-links/src/tabs/ApplicationTabs.jsx]: Updated this to use shouldComponentUpdate and added a page to manually trigger links length.
Commit: https://github.com/cerner/terra-framework/pull/365/commits/0f1d2006e8e545605f24a8d0c23adca7cf853f19
Test Page: https://terra-framework-deploye-pr-365.herokuapp.com/#/tests/terra-application-links/application-links/application-tabs-length-change

File: packages/terra-application-links/src/terra-dev-site/test/application-links/ApplicationTabsLengthChange.test.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    };
+  }
+
+  handleToggle() {
+    this.setState(prevState => ({ shortLinks: !prevState.shortLinks }));
+  }
+
+  render() {
+    return (
+      <Base locale="en-US" customMessages={messages}>
```
Comment (emilyrohrbough) [inline packages/terra-application-links/src/terra-dev-site/test/application-links/ApplicationTabsLengthChange.test.jsx]: The test page is already wrapped in Base. By adding it here, its nesting i18n-provides and preventing translations switching for this test page. Is it possible to remove it from this example?

Comment (bjankord) [reply packages/terra-application-links/src/terra-dev-site/test/application-links/ApplicationTabsLengthChange.test.jsx]: Update: https://github.com/cerner/terra-framework/pull/365/commits/98ceed58574d727fa9a2044985bc6b348873c529

**Pull request merged**